### ISSUE TITLE
`require 'hop'.hint_char1({direction = require 'hop.hint'.HintDirection.BEFORE_CURSOR})` errors when standing in empty line

### ISSUE THREAD

**Issue by FelipeLema** [smoka7/hop.nvim#36]
**Describe the bug**

hop.nvim errors when doing `require 'hop'.hint_char1({direction = require 'hop.hint'.HintDirection.BEFORE_CURSOR})`

**To Reproduce**

- Create empty buffer
- input the following text
```
...

```
- `:norm G` (make sure you're standing at the last line)
- `:lua require 'hop'.hint_char1({direction = require 'hop.hint'.HintDirection.BEFORE_CURSOR})` and input `.`
- currently getting  the following error
```
E5108: Error executing lua: .../user/.local/share/nvim/lazy/hop.nvim/lua/hop/init.lua:139: Invalid 'end_col': out of range
stack traceback:
	[C]: in function 'nvim_buf_set_extmark'
	.../user/.local/share/nvim/lazy/hop.nvim/lua/hop/init.lua:139: in function 'set_unmatched_lines'
	.../user/.local/share/nvim/lazy/hop.nvim/lua/hop/init.lua:155: in function 'apply_dimming'
	.../user/.local/share/nvim/lazy/hop.nvim/lua/hop/init.lua:364: in function 'hint_with_callback'
	.../user/.local/share/nvim/lazy/hop.nvim/lua/hop/init.lua:315: in function 'hint_with'
	.../user/.local/share/nvim/lazy/hop.nvim/lua/hop/init.lua:519: in function 'hint_char1'
	/home/user/.config/nvim/lua/custom/plugins/init.lua:29: in function </home/user/.config/nvim/lua/custom/plugins/init.lua:25>
```

**Expected behavior**

hop offers candidates to jump to any of the three `.`s in the first line

**Screenshots**
NA

**version (please complete the following information):**
 - Nvim version: 0.9.1
 - hop.nvim version: 877e175

**Additional context**
NA

**Comment by smoka7** [comment]
Can't reproduce it on 2d80315010389ab140a70a516c9817f263670cf4 update to the head.

**Comment by FelipeLema** [comment]
updated and problem is solved in current head (  2d80315 )

thanks
